src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -226,12 +234,10 @@ public async Task Put()
 
                 if (LoggingSource.AuditLog.IsInfoEnabled)
                 {
-                    var permissions = certificate?.Permissions != null
-                        ? Environment.NewLine + string.Join(Environment.NewLine, certificate.Permissions.Select(kvp => kvp.Key + ": " + kvp.Value.ToString()))
-                        : string.Empty;
+                    var permissions = FormatPermissions(certificate);
 
                     LogAuditFor("Certificates", "ADD", 
-                        $"New certificate {certificate?.Name} ['{certificate?.Thumbprint}']. Security Clearance: {certificate?.SecurityClearance}. Permissions:{permissions}.");
+                        $"New certificate {certificate?.Name} ['{certificate?.Thumbprint}']. Security Clearance: {certificate?.SecurityClearance}. Permissions:{permissions}. TwoFactor: {string.IsNullOrEmpty(twoFactorAuthenticationKey)}");
```

Comment:
This is a different endpoint than POST. I think this is where we upload an existing client cert right? If so, I'd say we should have a different message or action (e.g. UPLOAD instaed of ADD).

Reply:
POST is used to generate. Changed action there, it should resolve name clash.